torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -54,6 +56,70 @@ def reset_table(self) -> None:
 kernel_side_table = KernelSideTable()
 
 
+###############################################################################
+# Mutation Tracker
+
+
+@dataclasses.dataclass
+class MutationInfo:
+    mutated: bool = False
+    used_in_unknown: bool = False
+
+
+# Super basic mutation tracking pass that tracks which inputs are used in stores
+# It bails if any of the inputs are used in non tl.load/tl.store positions.
+class MutationTracker(ast.NodeVisitor):
+    def __init__(self, infos) -> None:
+        super().__init__()
+        self.infos = infos
+        self.in_load = False
+        self.in_store = False
+
+    def visit_Name(self, node):
+        if node.id not in self.infos:
+            return
+        if self.in_load:
+            pass
+        elif self.in_store:
+            self.infos[node.id].mutated = True
+        else:
+            self.infos[node.id].used_in_unknown = True
+
+    def visit_Call(self, node):
+        if (
+            isinstance(node.func, ast.Attribute)
+            and isinstance(node.func.value, ast.Name)
+            and node.func.value.id == "tl"
```

Comment:
What if `triton.language` is imported not as `tl`? Can we make this more robust by checking the actual module behind it? Most of the time it's `tl`, but I feel non-`tl` is a feasible case.